sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -496,6 +523,10 @@ def _get_bigquery_client():
             {{ feature }} as {{ featureview.name }}__{{ feature }}{% if loop.last %}{% else %}, {% endif %}
         {% endfor %}
     FROM {{ featureview.table_subquery }}
+    WHERE {{ featureview.event_timestamp_column }} <= '{{max_timestamp}}'
+    {% if featureview.ttl == 0 %}{% else %}
+    AND {{ featureview.event_timestamp_column }} >= Timestamp_sub('{{min_timestamp}}', interval {{ featureview.ttl }} second)
```

Comment:
I don't feel super strongly, but subtracting the TTL outside of the query could make this query simpler than doing it within. Happy to ship as is though.

Reply:
So, I feel that might overcomplicate the logic. The way I see it, each feature view has its own TTL which would mean we would need to pass a min_timestamp per feature_view if we wanted to subtract the TTL outside the query. However, if you have a cleaner way of doing it in mind, I don't mind making the modification!

Reply:
Ho I see what you mean. The current approach sounds fine to me to be honest